### DISCUSSION TITLE
Maybe _mise_hook should be called at the end of the activate content?

### DISCUSSION THREAD

**Discussion by AndydeCleyre** [jdx/mise#3260] [Ideas]
I'm using Zsh, and found that when I replaced otherwise-installed tools with mise-installed tools, such as cargo-installed `eza` with mise-installed `ubi:eza-community/eza`, the path would not include the relevant folder by the time I check for that tool's presence in later `.zshrc` code.

For example, if after loading up mise in my `.zshrc` I have:

```zsh
if (( $+commands[eza] )) {
  alias ls="eza --binary --octal-permissions --no-permissions --git --icons=always"
} else {
  alias ls="=ls --color=auto"
}
```

then the second `alias` will be used.

I found I could fix this by prefacing that code with:

```zsh
if (( $+functions[_mise_hook] ))  _mise_hook
```

forcing the PATH to be updated before the check.

Or, since I write the output of `mise activate zsh` to a file and source that, adding `_mise_hook` as a final line to that file also works.

Would it make sense to include that in the official `mise activate` content, so that tools which ought to be in the PATH regardless of current folder can be made available during shell startup?

**Comment by jdx** [comment]
I don't recall the reasons but this was the behavior for a while but I removed it because it caused more problems than it solved

**Comment by AndydeCleyre** [comment]
It looks like this is happening now, with https://github.com/jdx/mise/pull/3373, right? Thanks!
